libraries/chain/abi_serializer.cpp

```diff
@@ -141,47 +149,47 @@ namespace eosio { namespace chain {
       variants.clear();
       action_results.clear();
 
-      for( const auto& st : abi.structs )
-         structs[st.name] = st;
+      for( auto&& st : abi.structs )
```

Comment:
Not 100% sure, but I don't think `auto&&` is required, `auto&` is probably enough.

Reply:
`auto&` is enough, but I think `auto&&` is clearer intent.

Reply:
Hum. `st` will be a lvalue anyways. `auto&& st = <expr>` would be useful only if <expr> was (or could be) a rvalue.

In this case `abi.structs` is a lvalue and I don't believe that just declaring the loop variable `auto&&` does much, but then again it doesn't hurt either so imho it is fine to leave as is.

Reply:
On second thought, `auto&` is likely clearer intent here.